homeassistant/components/modbus/climate.py

```diff
@@ -413,8 +421,8 @@ async def async_set_swing_mode(self, swing_mode: str) -> None:
     async def async_set_temperature(self, **kwargs: Any) -> None:
         """Set new target temperature."""
         target_temperature = (
-            float(kwargs[ATTR_TEMPERATURE]) - self._offset
-        ) / self._scale
+            float(kwargs[ATTR_TEMPERATURE]) - self._target_temp_offset
+        ) / self._target_temp_scale
```

Comment:
Do you validate for target/current scale being configured to 0.

It does not make sense, but it is possible.